src/psyclone/psyir/frontend/fparser2.py

```diff
@@ -2218,6 +2219,39 @@ def process_declarations(self, parent, nodes, arg_list,
                     # Restore the fparser2 parse tree
                     node.children[2].items = tuple(orig_children)
 
+            elif isinstance(node, Fortran2003.Common_Stmt):
+
+                # If the first common object has a name, use it
+                if node.children[0][0][0]:
+                    name = node.children[0][0][0]
+                else:
+                    name = "unnamed"
+
+                # Place the declaration statement into a UnknownFortranType
+                # (for now we just want to reproduce it). The name of the
+                # codeblock is not in the same namespace as the variable
+                # symbols names, so we add a psyclone prefix to
+                # differentiate them.
+                parent.symbol_table.new_symbol(
+                    f"_PSYCLONE_COMMONBLOCK_{name}",
+                    symbol_type=DataSymbol,
+                    datatype=UnknownFortranType(str(node)))
+
+                # Get the names of the symbols accessed with the commonblock,
+                # they are already defined in the symbol table but they must
+                # now have a common-block interface.
+                try:
+                    for cb_object in node.children[0]:
+                        for symbol_name in cb_object[1].items:
```

Comment:
I added a test for this. The current implementation does the right think already because the "D (10, 30)" lookup doesn't finds it which produces a NotImplementedError that ends up putting the whole routine in a CodeBlock.

Reply:
Thanks.